packages/api/lambdas/write-db-dlq-records-to-s3.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+const get = require('lodash/get');
+const { s3PutObject, s3ObjectExists } = require('@cumulus/aws-client/S3');
+const { parseSQSMessageBody } = require('@cumulus/aws-client/SQS');
+
+async function handler(event, _) {
+  if (!process.env.system_bucket) throw new Error('System bucket env var is required.');
+  if (!process.env.stackName) throw new Error('Could not determine archive path as stackName env var is undefined.');
+  const sqsMessages = get(event, 'Records', []);
+  /* eslint-disable no-restricted-syntax, no-await-in-loop */
+  for (const message of sqsMessages) {
+    const cumulusMessage = parseSQSMessageBody(message);
+    const executionName = cumulusMessage.cumulus_meta.execution_name;
+    // version messages as workflows can produce multiple messages that may all fail.
+    let s3IdVersionSuffix = 1;
+    let s3Identifier = `${executionName}-${s3IdVersionSuffix}`;
+    while (await s3ObjectExists({
```

Comment:
Correct. If both messages failed due to a database outage and ended up on the DLQ I figured we would want to capture both, rather than extend the database write logic into this function where it picks and chooses which is the more 'valid' record (which would bring more issues with malformed records, and would still necessarily include an 'exists' check).

Reply:
Refactored out of existence.